### PR TITLE
feat: switch native calls to native events

### PR DESCRIPTION
**Pull request by osdnk**
you know 

# Summary

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [ ] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: ios/ReactNativePageView.h [hunk 1]
```diff
@@ -7,8 +7,11 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ReactNativePageView: UIView <UIPageViewControllerDataSource, UIPageViewControllerDelegate,UIScrollViewDelegate>
 
+- (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher;
+
 @property(strong, nonatomic, readonly) UIPageViewController *reactPageViewController;
 @property(strong, nonatomic, readonly) UIPageControl *reactPageIndicatorView;
+@property(strong, nonatomic, readonly) RCTEventDispatcher *eventDispatcher;
```
Comment (ptrocki) [inline ios/ReactNativePageView.h]: WDYT about make this property weak ? 

https://github.com/facebook/react-native/blob/84f5ebe4f941e752ba96b4d4a75dc6c6f5faa665/React/Base/RCTTouchHandler.m#L29

https://github.com/facebook/react-native/blob/8131b7bb7b4794e0e7003a6e3d34e1ebe4b8b9bc/Libraries/Text/TextInput/RCTBaseTextInputView.m#L25

Comment (ptrocki) [reply ios/ReactNativePageView.h]: Ohh weakSelf is used https://github.com/react-native-community/react-native-viewpager/pull/46/files#diff-42352f0da98b553d1ace487a18f6d823R363. I close this comment.

File: ios/ReactNativePageView.m [hunk 2]
```diff
@@ -1,11 +1,216 @@
...
+
+- (NSArray *)arguments
+{
+    return @[self.viewTag, RCTNormalizeInputEventName(self.eventName), @{
+                 @"position": _position,
+                 @"offset": _offset
+                 }];
+}
+
+- (id<RCTEvent>)coalesceWithEvent:(id<RCTEvent>)newEvent;
```
Comment (dratwas) [inline ios/ReactNativePageView.m]: This is optional and needs to be implemented only if `canCoalesce` returns `YES`

File: ios/ReactNativePageView.m [hunk 3]
```diff
@@ -1,11 +1,216 @@
...
+                           offset:(NSNumber *)offset
+                    coalescingKey:(uint16_t)coalescingKey;
+
+@end
+
+@implementation RCTOnPageScrollEvent
+{
+    NSNumber* _position;
+    NSNumber* _offset;
+    uint16_t _coalescingKey;
```
Comment (dratwas) [inline ios/ReactNativePageView.m]: Do we really need this coalescingKey?

**Comment by osdnk** [comment]
not really
image

File: ios/ReactNativePageView.h (hunk 1 above)
Comment (osdnk) [reply ios/ReactNativePageView.h]: yes, sorry!

File: ios/ReactNativePageView.m (hunk 3 above)
Comment (osdnk) [reply ios/ReactNativePageView.m]: Yes, it's not obvious for be why, but we have to implement it always

Comment (dratwas) [reply ios/ReactNativePageView.m]: Nvm, I checked it on `react-native` master branch :) It is optional from version `0.60` Sorry :D

**Comment by osdnk** [comment]
native driversss!
https://streamable.com/c4kma

**Comment by ferrannp** [comment]
That is dope @osdnk !!!

**Pull request merged**